### ISSUE TITLE
Unable To Make Offer

### ISSUE THREAD

**Issue by micahalcorn** [OriginProtocol/origin#3568]
I'm still working on identifying what variables are in play here. At least the following is true:

Buyer: `0x522731a061e896B5Db9dDff9234fB5461A533710`
Seller: `0x96FEb7b6F808dd2BBd09c9E5ccdE77caBd58d019`
Network: Rinkeby
Listing: [4-001-6](https://dapp.staging.originprotocol.com/#/listing/4-001-6) (Housing)
Error: "Cannot read property 'isBefore' of undefined" on `getAvailability`

-----

At Origin, we use [ZenHub](https://www.zenhub.com/) to manage our engineering tasks and product development. [Download their browser extension](https://www.zenhub.com/extension) and check out our open workspace at [github.com/originprotocol/origin#zenhub](https://github.com/originprotocol/origin#zenhub).

**Comment by micahalcorn** [comment]
Services listing seems to work fine. ☑️

**Comment by micahalcorn** [comment]
I can't seem to reproduce this, but it was last seen in Sentry seven hours ago. 🙁 Have we shipped something to production today that could have fixed this, or is there any other reason to believe that it's a non-issue @nick?

**Comment by tomlinton** [comment]
Relevant excerpt:

```
  getAvailability(startStr, endStr) {
    let start = typeof startStr === 'string' ? dayjs(startStr) : startStr
    let end = typeof endStr === 'string' ? dayjs(endStr) : endStr
    const days = []
    if (end.isBefore(start)) {
      const newEnd = start
      start = end
      end = newEnd
    }
```

It's the end date, so somehow it's getting an undefined value. Possible scenarios are endStr is `undefined` in the functional call or `dayjs(endStr)` is returning undefined (non ISO 8601 date string perhaps?).

**Comment by shahthepro** [comment]
@micahalcorn I couldn't reproduce this either on the same listing. Is this still happening?

**Comment by micahalcorn** [comment]
Last seen 17-18 days ago 🤷‍♂️

**Comment by micahalcorn** [comment]
Seen just now on [this listing](https://dapp.staging.originprotocol.com/#/listing/4-000-209) when selecting 10/22 (tomorrow) during 19:07 PDT 😩

**Comment by tomlinton** [comment]
I made an offer fine with the parameters below.

<img width="706" alt="Screen Shot 2019-10-22 at 3 16 56 PM" src="https://user-images.githubusercontent.com/489202/67255568-28266500-f4df-11e9-85d9-f66dadc158a4.png">

I think what is important here is that @micahalcorn is still in a timezone where it is the 21st, and for me (and probably more importantly UTC) it is now the 22nd.

**Comment by shahthepro** [comment]
Played around with this. It happens when you select a start date but not an end date. I was not able to reproduce it initially since I selected two date by instinct. 

The "Save" button should be disabled if both dates are not selected. Looks like I somehow messed up the constraint that disables the button. Will include the fix in `shahthepro/bug-fixes` branch

**Comment by micahalcorn** [comment]
I'm fairly certain that I was selecting an end date when I reported this. I may be wrong, so we'll consider it resolved by d277d2e5634fae2d45ad15364dfbbc6dd959e5d7 until we see it again. 🤞
